sdk/src/main/java/com/sygic/travel/sdk/auth/facade/AuthFacade.kt

```diff
@@ -0,0 +1,170 @@
+package com.sygic.travel.sdk.auth.facade
+
+import com.sygic.travel.sdk.Callback
+import com.sygic.travel.sdk.auth.model.AuthorizationRequest
+import com.sygic.travel.sdk.auth.model.UserRegistrationRequest
+import com.sygic.travel.sdk.auth.model.UserRegistrationResponse
+import com.sygic.travel.sdk.auth.model.UserSession
+import com.sygic.travel.sdk.auth.service.AuthService
+import com.sygic.travel.sdk.auth.service.AuthStorageService
+import com.sygic.travel.sdk.utils.runAsync
+import com.sygic.travel.sdk.utils.runWithCallback
+
+
+class AuthFacade(
+	private val authService: AuthService,
+	private val authStorageService: AuthStorageService
+) {
+
+	companion object {
+		const val CLIENT_ID = "sygictravel_android_sdk_demo"
+		const val DEVICE_PLATFORM = "android"
+	}
+
+	fun loginUserByPassword(
+		username: String,
+		password: String,
+		callback: Callback<UserSession?>
+	) {
+		val request = RequestBuilder.passwordAuthRequest(username, password)
+		runWithCallback({ authService.authorize(request) }, callback)
+	}
+
+	suspend fun loginUserByPassword(username: String, password: String): UserSession? {
+		val request = RequestBuilder.passwordAuthRequest(username, password)
+		return runAsync { authService.authorize(request) }
+	}
+
+	fun loginUserByGoogle(googleToken: String?, callback: Callback<UserSession?>) {
+		val request = RequestBuilder.googleAuthRequest(googleToken)
+		runWithCallback({ authService.authorize(request) }, callback)
+	}
+
+	suspend fun loginUserByGoogle(googleToken: String?): UserSession? {
+		val request = RequestBuilder.googleAuthRequest(googleToken)
+		return runAsync { authService.authorize(request) }
+	}
+
+	fun loginUserByFacebook(facebookToken: String?, callback: Callback<UserSession?>) {
+		val request = RequestBuilder.facebookAuthRequest(facebookToken)
+		runWithCallback({ authService.authorize(request) }, callback)
+	}
+
+	suspend fun loginUserByFacebook(facebookToken: String?): UserSession? {
+		val request = RequestBuilder.facebookAuthRequest(facebookToken)
+		return runAsync { authService.authorize(request) }
+	}
+
+	fun loginUserByDeviceId(callback: Callback<UserSession?>) {
+		val authRequest = RequestBuilder.deviceIdAuthRequest(authStorageService.getDeviceId())
+		runWithCallback({ authService.authorize(authRequest) }, callback)
+	}
+
+	suspend fun loginUserByDeviceId(): UserSession? {
+		val authRequest = RequestBuilder.deviceIdAuthRequest(authStorageService.getDeviceId())
+		return runAsync { authService.authorize(authRequest) }
+	}
+
+	fun loginUserByJwtToken(jwtToken: String, callback: Callback<UserSession?>) {
+		val authRequest = RequestBuilder.jwtTokenAuthRequest(jwtToken)
+		runWithCallback({ authService.authorize(authRequest) }, callback)
+	}
+
+	suspend fun loginUserByJwtToken(jwtToken: String): UserSession? {
+		val authRequest = RequestBuilder.jwtTokenAuthRequest(jwtToken)
+		return runAsync { authService.authorize(authRequest) }
+	}
+
+	fun registerUser(
```

Comment:
I wouldn't wrap

Reply:
This one should be wrapped, because unwrapped it exceeds 100 chars. But I unwrapped many that does not.